Jenkinsfile

```diff
@@ -10,76 +10,17 @@ pipeline {
-                    env.JAVA_HOME = tool('Oracle JDK 8')
-                }
-
-                sh returnStdout: true, script: 'cleanup.sh'
-            }
-        }
-
-        stage('Checkout') {
-            steps {
-                checkout scm
-            }
-        }
-
-        stage('Build') {
-            steps {
-                configFileProvider([configFile(fileId: 'maven-settings-with-deploy-snapshot', variable: 'MAVEN_SETTINGS')]) {
-                    sh "$MAVEN_HOME/bin/mvn clean install -B -V -e -s $MAVEN_SETTINGS -DskipTests"
-                }
-                warnings canRunOnFailed: true, consoleParsers: [[parserName: 'Maven'], [parserName: 'Java Compiler (javac)']], shouldDetectModules: true
-                checkstyle canRunOnFailed: true, pattern: '**/target/checkstyle-result.xml', shouldDetectModules: true
-            }
-        }
-
-        stage('Tests') {
-            steps {
-                configFileProvider([configFile(fileId: 'maven-settings-with-deploy-snapshot', variable: 'MAVEN_SETTINGS')]) {
-                    sh "$MAVEN_HOME/bin/mvn verify -B -V -e -s $MAVEN_SETTINGS -Dmaven.test.failure.ignore=true -Dansi.strip"
-                }
-                // TODO Add StabilityTestDataPublisher after https://issues.jenkins-ci.org/browse/JENKINS-42610 is fixed
-                // Capture target/surefire-reports/*.xml, target/failsafe-reports/*.xml,
-                // target/failsafe-reports-embedded/*.xml, target/failsafe-reports-remote/*.xml
-                junit testResults: '**/target/*-reports*/*.xml',
-                        testDataPublishers: [[$class: 'ClaimTestDataPublisher']],
-                        healthScaleFactor: 100, allowEmptyResults: true
-
-                // Workaround for SUREFIRE-1426: Fail the build if there a fork crashed
-                script {
-                    if (manager.logContains("org.apache.maven.surefire.booter.SurefireBooterForkException:.*")) {
-                        echo "Fork error found"
-                        manager.buildFailure()
-                    }
-                }
-
-                // Dump any dump files to the console
-                sh 'find . -name "*.dump*" -exec echo {} \\; -exec cat {} \\;'
-                sh 'find . -name "hs_err_*" -exec echo {} \\; -exec grep "^# " {} \\;'
+                sh "fail"
             }
         }
     }
 
     post {
-        always {
-            // Archive logs and dump files
-            sh 'find . \\( -name "*.log" -o -name "*.dump*" -o -name "hs_err_*" \\) -exec xz {} \\;'
-            archiveArtifacts allowEmptyArchive: true, artifacts: '**/*.xz,documentation/target/generated-html/**'
-
-            // Clean
-            sh 'git clean -fdx -e "*.hprof" || echo "git clean failed, exit code $?"'
+       failure {
+            emailext to: '${DEFAULT_RECIPIENTS}', subject: '${DEFAULT_SUBJECT}', 
+            body: '${DEFAULT_CONTENT}'
```

Comment:
Is it possible to include a link to the job that failed? Or is it included automatically?

Reply:
@tristantarrant yeah I had removed the actual build to test just the email stuff. Now everything should in in place.

@slaskawi with the default subject/body (defined in jenkins/configuration) mail looks like this:

    Subject: CppRelease » Nightly CPP - Build # 62 - Failure!
    Body: CppRelease » Nightly CPP - Build # 62 - Failure: Check console output at https://ci.infinispan.org/job/CppRelease/job/Nightly%20CPP/62/ to view the results.

We can add more info if needed.